tests/lru_cache_test.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+import string
+import tempfile
+
+from absl.testing import absltest
+
+from jax._src.lru_cache import LRUCacheImpl
+import jax._src.test_util as jtu
+
+
+class LRUCacheTest(jtu.JaxTestCase):
```

Comment:
Consider adding a test with multiple processes modifying the same cache directory.

Reply:
This is kind of tricky, maybe it should be a follow up?

Reply:
Sure, sounds good.

Reply:
Done by adding a TODO in the tests